addons/mail/tests/discuss/test_discuss_channel_member.py

```diff
@@ -28,133 +28,32 @@ def setUpClass(cls):
-            channel_member_1.with_user(self.user_2).partner_id = self.user_2.partner_id
-        self.assertEqual(channel_member_1.partner_id, self.user_1.partner_id)
-
-        # Even with SUDO, partner_id of channel.member should not be changed.
+        cls.group = cls.env["discuss.channel"].create({"name": "Group", "channel_type": "group"})
+        cls.group.channel_member_ids.unlink()
+
+    def test_cannot_change_member_immutable_fields(self):
+        channel = self.env["discuss.channel"]._create_channel(group_id=None, name="General")
+        bob = new_test_user(self.env, "bob", groups="base.group_user")
+        another_channel = self.env["discuss.channel"]._create_channel(group_id=None, name="Another channel")
+        another_partner = self.env["res.partner"].create({"name": "John"})
+        guest = self.env["mail.guest"].create({"name": "Jane"})
+        member = channel._add_members(users=bob)
         with self.assertRaises(AccessError):
-            channel_member_1.with_user(self.user_2).sudo().partner_id = self.user_2.partner_id
-
-    def test_group_members(self):
-        """Test invitation in group part 1 (invite using crud methods)."""
-        self.group.with_user(self.user_1).sudo()._add_members(users=self.user_1)
-        channel_members = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id)])
-        self.assertEqual(len(channel_members), 1)
-
-        # User 2 is not in the group, they can not invite user 3
-        with self.assertRaises(AccessError):
-            self.env['discuss.channel.member'].with_user(self.user_2).create({
-                'partner_id': self.user_portal.partner_id.id,
-                'channel_id': self.group.id,
-            })
-
-        # User 1 is in the group, they can invite other users
-        self.env['discuss.channel.member'].with_user(self.user_1).create({
-            'partner_id': self.user_portal.partner_id.id,
-            'channel_id': self.group.id,
-        })
-        channel_members = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id)])
-        self.assertEqual(channel_members.mapped('partner_id'), self.user_1.partner_id | self.user_portal.partner_id)
-
-        # But User 3 can not write on the `discuss.channel.member` of other user
-        channel_member_1 = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id), ('partner_id', '=', self.user_1.partner_id.id)])
-        channel_member_3 = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id), ('partner_id', '=', self.user_portal.partner_id.id)])
-        channel_member_3.with_user(self.user_portal).custom_channel_name = 'Test'
+            member.channel_id = another_channel
         with self.assertRaises(AccessError):
-            channel_member_1.with_user(self.user_2).custom_channel_name = 'Blabla'
-        self.assertNotEqual(channel_member_1.custom_channel_name, 'Blabla')
-
-    def test_group_invite(self):
```

Comment:
This is kind of redundant with the other tests, except that it checks that portal user can be invited by a member of the group.

 It's not worth maintaining this test: we already check that a user is allowed to invite others. There are other tests such as `TestMailPublicPage` that ensure that portal users can join channels. And finally, `test_01_discuss_channel_access` ensures portal don't see more than they should, even when they are members.